ci/jenkins/jobs/macros.yaml

```diff
@@ -283,7 +283,7 @@
           #!/bin/bash
           set -e
           DOCKER_REGISTRY="$(head -n1 ci/docker-registry)"
-          ./ci/jenkins/test.sh --testcase e2e --registry ${{DOCKER_REGISTRY}} --testbed-type jumper
+          ./ci/jenkins/test.sh --testcase e2e --registry ${DOCKER_REGISTRY} --testbed-type jumper
```

Comment:
I'm really confused. https://github.com/antrea-io/antrea/pull/5464 changed it to double brackets to "fix" it and it seemed passing the validation. Now it's reverted to single bracket to "fix" it again and it also can pass the validation? Is the validation really useful?
Can you share what's the rule of using single bracket or double brackets or any doc about this?

Reply:
Yes, in PR #5455 we introduced a new Jenkins variable "conformance_type", which is only used in macro `builder-conformance-jumper`, so we only need to use double brackets in job `conformance`.  I remember that I removed the double brackets in e2e job, but it still exists(Maybe VSCode automatically save did work). So I created a new PR to remove it. 

As for the Jenkins jobs validation, it used `jenkins-job test` command to test the Jenkins job, it's a Jenkins issue that if we do not use Jenkins variable in one job, and still use double brackets for the shell variables, the test still passes. 

As for the rule of using a single bracket or double brackets, if we introduce a Jenkins variable in a job, such  `conformance_type`,then we need to use double brackets, if not, then single brackets. For example,  in the following Jenkins job, the builder uses variable `conformance_type` (without dollar symbol), then we need to use double brackets for the shell variable `DOCKER_REGISTRY`. 

```yaml
- builder:
    name: builder-rancher-conformance
    builders:
      - shell: |-
          #!/bin/bash
          set -ex
          DOCKER_REGISTRY="$(head -n1 ci/docker-registry)"
          chmod +x ./ci/jenkins/test-rancher.sh
          ./ci/jenkins/test-rancher.sh --cluster-name rancher-test --testcase '{conformance_type}' --registry ${{DOCKER_REGISTRY}}
```

Reply:
Thanks for the explanation. It's much clearer to me now.